File: test_script/bench/bench_fn_fib.rb

# def fib(n):
#     if n == 1 or n == 2:
#         return 1
#     else:
#         return fib(n - 1) + fib(n - 2)

# for i in range(18000000):
#     fib(2)

def fib(n)
    return 1 if n == 1 || n == 2
    fib(n - 1) + fib(n - 2)
end

18_000_000.times do
    fib(2)
end

File: test_script/bench/bench_fn_fib.fs
fn fib(n) {
    if n == 1 or n == 2 {
        return 1
    } else {
        return fib(n - 1) + fib(n - 2)
    }
}

for i in 0..3000000 {
    fib(2)
}
